### PR TITLE
add ability to inspect version to all binaries

### PR DESCRIPTION
**Pull request by jtolds**
Currently, it is hard to tell if a release binary we built was built correctly. This commit makes it so running binaries with the `version` subcommand will output build version information, and the debug port now has a new /version handler that returns version information. Last, this updates the release script to allow builds without a version tag, but will not allow the release flag to be set in that case.

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: internal/version/version.go [hunk 1]
```diff
@@ -35,6 +36,9 @@ type Info struct {
 	CommitHash string    `json:"commitHash,omitempty"`
 	Version    SemVer    `json:"version"`
 	Release    bool      `json:"release,omitempty"`
+
+	crcOnce       sync.Once
+	commitHashCRC uint32
```
Comment (egonelbre) [inline internal/version/version.go]: Any reason we are not making it part of build flags?

File: internal/version/stats.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package version
+
+import "hash/crc32"
+
+func (v *Info) Stats(cb func(name string, val float64)) {
```
Comment (egonelbre) [inline internal/version/stats.go]: Use `reportValue` or similar. When reading the code `cb` confused me.

Comment (jtolds) [reply internal/version/stats.go]: callback? but okay yeah

File: internal/version/version.go (hunk 1 above)
Comment (jtolds) [reply internal/version/version.go]: uhh, yeah i mean, from a precomputation perspective that will save every process some cycles, but from a correctness perspective it adds an avenue for bugs. this way (don't let it be configured) means we're much more unlikely to get it wrong, and i think correctness here is more important than the perf gain we get from basically denormalizing and repeating the data

File: internal/version/version.go [hunk 3]
```diff
@@ -153,4 +157,5 @@ func init() {
 	if Build.Timestamp.Unix() == 0 || Build.CommitHash == "" {
 		Build.Release = false
 	}
+
```
Comment (stefanbenten) [inline internal/version/version.go]: Besides this empty line, it looks good 👍

**Pull request merged**